actionview/lib/action_view/renderer/template_renderer.rb

```diff
@@ -64,13 +64,14 @@ def render_template(view, template, layout_name, locals)
 
       def render_with_layout(view, template, path, locals)
         layout  = path && find_layout(path, locals.keys, [formats.first])
-        content = yield(layout)
 
         body = if layout
-          view.view_flow.set(:layout, content)
-          layout.render(view, locals) { |*name| view._layout_for(*name) }
+          ActiveSupport::Notifications.instrument("render_template.action_view", identifier: layout.identifier) do
+            view.view_flow.set(:layout, yield(layout))
+            layout.render(view, locals) { |*name| view._layout_for(*name) }
+          end
         else
-          content
+          yield(layout)
```

Comment:
If this works (as far as I can tell, it does), imo this makes the flow clearer:

```suggestion
          yield
```

Reply:
still prefer to do this if it works